### PR TITLE
Link MOSAIC nitrate and MAM5 strat sulfate treatment

### PR DESCRIPTION
**Pull request by mingxuanwupnnl**
This PR provides the capability of running MOSAIC nitrate and MAM5 stratospheric sulfate treatment together. This PR is a stealth feature which does not turn on MOSAIC and MAM5 by default.

Two conditional compilation if statements: MODAL_AERO_5MODE and MOSAIC_SPECIES

The code modifications are mainly for:

1. Moving conditional compilation if statements MODAL_AERO_5MODE together with MODAL_AERO_4MODE_MOM, because MAM5 only adds a strat coarse mode compared to MAM4. Minimize the code changes so that only 

( MODAL_AERO_4MODE_MOM || MODAL_AERO_5MODE ) && MOSAIC_SPECIES 
and 
( MODAL_AERO_4MODE_MOM || MODAL_AERO_5MODE )
work for most places.

2. Adding chem_mech.in files for chemUCI-MOSAIC and chemUCI-MOSAIC-MAM5.

Test suite results
https://acme-climate.atlassian.net/wiki/spaces/NGDAP/pages/3523149825/Couple+MOSAIC+and+MAM5+strat+sulfate 

It is BFB (stealth) for chemUCI only simulations.
test suites results compared with NGD_v3atm baseline (./create_test $TESTNAMES --compare -b NGD_v3atm):
/lcrc/group/e3sm/ac.mwu/scratch/chrys/

PASS
SMS_Ln5.ne30pg2, 
SMS_Ln.ne4pg2
ERS_Ln11.ne30pg2, 
ERS_Ln9.ne4pg2, 
ERS_ln11_P512x1.ne30.pg2
REP_Ln9 (FAIL for TPUTCOMP Error: TPUTCOMP: Computation time increase > 10% from baseline)
PET_Ln5_P256x2.ne30pg2

FAIL
PET_Ln5.ne30pg2
PET_Ln5.ne4pg2
PEM_Ln5.ne4pg2 (FAIL RUN time; PEND COMPARE_base_modpes)
PEM_Ln9.ne30pg2

It is BFB (stealth) for chemUCI-MOSAIC simulations.

It is not restart-BFB for chemUCI-MOSAIC-MAM5, and 1x10 and 2x5 test runs for NGD_v3atm branch also show non restart-BFB for MAM5 only simulations.

### CODE REVIEW

File: components/eam/chem_proc/inputs/pp_chemUCI_linozv3_mam4_resus_mom_soag_mosaic_tag.in [hunk 1]
```diff
@@ -0,0 +1,300 @@
+Comments
```
Comment (hwangacme) [inline components/eam/chem_proc/inputs/pp_chemUCI_linozv3_mam4_resus_mom_soag_mosaic_tag.in:1]: @mingxuanwupnnl, Why is this file needed for the coupling between MOSAIC and MAM5?

File: components/eam/src/chemistry/modal_aero/aero_model.F90 [hunk 2]
```diff
@@ -856,6 +854,50 @@ subroutine aero_model_init( pbuf2d, species_class, iflagaa )
     index_tot_mass(4,3) = get_spc_ndx('mom_a4')
     index_chm_mass(4,1) = get_spc_ndx('bc_a4' )
     !
+#elif ( ( defined MODAL_AERO_4MODE_MOM ) && ( defined RAIN_EVAP_TO_COARSE_AERO ) )
```
Comment (hwangacme) [inline components/eam/src/chemistry/modal_aero/aero_model.F90:857]: Why is this new block of code needed (for the default aerosol scheme 4MODE_MOM)?

File: components/eam/src/chemistry/modal_aero/aero_model.F90 [hunk 3]
```diff
@@ -894,8 +936,124 @@ subroutine aero_model_init( pbuf2d, species_class, iflagaa )
...
+    index_chm_mass(3,1)  = get_spc_ndx('so4_a3')
+    index_chm_mass(3,2)  = get_spc_ndx('soa_a3')
+    index_chm_mass(3,3)  = get_spc_ndx('bc_a3')
+    !
+    ! POM mode
+    !
+    index_tot_mass(4,1) = get_spc_ndx('pom_a4')
+    index_tot_mass(4,2) = get_spc_ndx('bc_a4')
+    index_tot_mass(4,3) = get_spc_ndx('mom_a4')
+    index_chm_mass(4,1) = get_spc_ndx('bc_a1' )
```
Comment (hwangacme) [inline components/eam/src/chemistry/modal_aero/aero_model.F90:1048]: Is "bc_a1" in this line a mistake? It is also in the default option two blocks above.

File: components/eam/chem_proc/inputs/pp_chemUCI_linozv3_mam4_resus_mom_soag_mosaic_tag.in (hunk 1 above)
Comment (mingxuanwupnnl) [reply components/eam/chem_proc/inputs/pp_chemUCI_linozv3_mam4_resus_mom_soag_mosaic_tag.in:1]: @hwangacme it is only for chemUCI-MOSAIC, I did not put chem_mech.in there for chemUCI-MOSAIC previously.

File: components/eam/src/chemistry/modal_aero/aero_model.F90 (hunk 2 above)
Comment (mingxuanwupnnl) [reply components/eam/src/chemistry/modal_aero/aero_model.F90:857]: Currently, this whole big chunk of code only matters for MAM5, as it use diagnostic diameter/surface area (SAD). If use read in SAD file (default for MAM4),  then the setting for index_tot_mass/index_chm_mass does not matter. Just put there for potential use?

Also, this 4MODE_MOM + RAIN_EVAP is missing (but it is the default for MAM4), so I add this part.

File: components/eam/src/chemistry/modal_aero/aero_model.F90 (hunk 3 above)
Comment (mingxuanwupnnl) [reply components/eam/src/chemistry/modal_aero/aero_model.F90:1048]: @hwangacme  Yes. It is. Will correct it.

File: components/eam/src/physics/cam/nucleate_ice_cam.F90 [hunk 4]
```diff
@@ -804,10 +804,9 @@ subroutine nucleate_ice_cam_calc( &
                   else
                      ! 3-mode -- needs weighting for dust since dust and seasalt
                      !           are combined in the "coarse" mode type
-#if (defined MODAL_AERO_4MODE_MOM && defined RAIN_EVAP_TO_COARSE_AERO )
-                     wght = dmc/(ssmc + dmc + so4mc + bcmc + pommc + soamc + mommc)
-!kzm ++
-#elif (defined MODAL_AERO_5MODE && defined RAIN_EVAP_TO_COARSE_AERO )
+#if ( ( defined MODAL_AERO_4MODE_MOM || defined MODAL_AERO_5MODE ) && ( defined RAIN_EVAP_TO_COARSE_AERO ) && ( defined MOSAIC_SPECIES ) )
+                     wght = dmc/(ssmc + dmc + so4mc + no3mc + nh4mc + bcmc + pommc + soamc + mommc)
```
Comment (hwangacme) [inline components/eam/src/physics/cam/nucleate_ice_cam.F90:808]: @mingxuanwupnnl , is chloride mass already included in the sea salt mass when MOSAIC is used?

Comment (mingxuanwupnnl) [reply components/eam/src/physics/cam/nucleate_ice_cam.F90:808]: @hwangacme Yes, see above lines (777). When MOSAIC is used, ssmc is the sum of coarse_nacl and coarse_cl.

**Review by hwangacme — APPROVED** [review]

**Review by keziming — CHANGES_REQUESTED** [review]

File: components/eam/src/chemistry/utils/prescribed_volcaero.F90 [hunk 5]
```diff
@@ -212,11 +212,6 @@ subroutine prescribed_volcaero_init(is_cmip6_volc)
     is_cmip6_volc = .false.
     if (trim(adjustl(file_type))== 'VOLC_CMIP6') then
        is_cmip6_volc = .true.
-#if (defined MODAL_AERO_5MODE)
-     write(iulog,*)'kzm_rescribed_volcaero_mam5 '
-     is_cmip6_volc = .false.
-#endif
-
```
[the remark this answers is not in the document]
Comment (mingxuanwupnnl) [reply components/eam/src/chemistry/utils/prescribed_volcaero.F90:219]: @keziming 
I removed that part because I found that this can be easily turned off just by setting
 
prescribed_volcaero_file = ‘’
prescribed_volcaero_datapath = ‘’

in the namelist.

It would be better than hardwiring is_cmip6_volc = .false. in prescribed_volcaero.F90
